tools/codegen/dest/lazy_ir.py

```diff
@@ -33,33 +33,37 @@ def node_ctor_arg_rvalue_string(arg: NamedCType, schema: LazyIrSchema) -> str:
         else:
             raise AssertionError("TODO not sure if there are other valid types to handle here")
     else:
-        if isinstance(arg.type, VectorCType) and isinstance(arg.type.elem, BaseCType):
-            return f"std::vector<{arg.type.elem.type}>({arg.name}.begin(), {arg.name}.end())"
-        elif (isinstance(arg.type, OptionalCType) and
-                isinstance(arg.type.elem, VectorCType) and
-                isinstance(arg.type.elem.elem, BaseCType)):
-            return f"torch::lazy::ToOptionalVector<{arg.type.elem.elem.type}>({arg.name})"
+        if isinstance(arg.lazy_type, VectorCType) and isinstance(arg.lazy_type.elem, BaseCType):
+            return f"std::vector<{arg.lazy_type.elem.type}>({arg.name}.begin(), {arg.name}.end())"
+        elif (isinstance(arg.lazy_type, OptionalCType) and
+                isinstance(arg.lazy_type.elem, VectorCType) and
+                isinstance(arg.lazy_type.elem.elem, BaseCType)):
+            return f"torch::lazy::ToOptionalVector<{arg.lazy_type.elem.elem.type}>({arg.name})"
         else:
             return f"{arg.name}"
 
 def node_ctor_inputs(schema: LazyIrSchema) -> str:
     """
     Produce a formatted string with the arguments as passed into the constructor of a node class.
     """
-    node_ctor_values = [node_ctor_arg_rvalue_string(arg, schema) for arg in schema.filtered_types()]
+    node_ctor_values = [node_ctor_arg_rvalue_string(arg) for arg in schema.filtered_args()]
     return ",\n                              ".join(node_ctor_values)
 
 def gen_fallback_code(schema: LazyIrSchema, overload_name: str) -> str:
     """
     Generate code that falls back to eager conditioned on a predicate
     """
-    fallback_args = ",\n                ".join([str(arg.name) for arg in schema.filtered_types()])
+    fallback_args = ",\n                ".join([str(arg.name) for arg in schema.filtered_args(generator=True)])
     if len(overload_name):
         aten_op_str = f"ATEN_OP2({schema.aten_name}, {overload_name})"
     else:
         aten_op_str = f"ATEN_OP({schema.aten_name})"
+    or_has_generator = ""
+    if schema.generator_arg:
+        # TODO(whc) generators are? always optional and there is never more than one
```

Comment:
@bdhirsh any idea if this is true?

Reply:
i just took a quick look at native_functions.yaml and it looks like it is.

Maybe just add an assert for it here though? That way, if (somehow) someone decides to add a non-optional generator in the future, when we fail with a nice codegen error message, instead of getting a bad error message at C++ compile time.